src/static-pages/index.tsx

```diff
@@ -0,0 +1,298 @@
+    </Template>
+  )
+}
+
+export function RevisionsOverview(revisions: RevisedPage[]) {
+  const current = revisions[0]
+  const title = `Versions: ${current.title}`
+
+  return (
+    <Template title={title} lang={current.lang}>
+      <h1>{title}</h1>
+      There are the following archived versions of {current.title} available:
+      <ul>
+        {revisions.map(rev => {
+          const link = `/${rev.revisedType}/archiv/${getRevisionId(rev)}`
+
+          return (
+            <li>
+              <a href={link}>
+                {rev.revision.toLocaleDateString(rev.lang)}
+                {rev.isCurrentRevision ? ' (current version)' : ''}
+              </a>
+            </li>
+          )
+        })}
+      </ul>
+    </Template>
+  )
+}
+
+export async function fetchContent<A extends Page>(
+  page: A
+): Promise<WithContent<A> | null> {
+  const response = await fetch(new Request(page.url))
+
+  if (response.ok) {
+    const text = await response.text()
+    const rawContent = page.url.endsWith('.md') ? markdownToHtml(text) : text
+    const sanitizedContent = sanitizeHtml(rawContent)
+    const content = sanitizedContent.replace(
+      '__JS_GOOGLE_ANALYTICS_DEACTIVATE__',
+      'javascript:gaOptout();'
+    )
+
+    return { ...page, content }
+  } else {
+    return null
+  }
+}
+
+export function findRevisionById<A extends RevisedSpec>(
+  revisions: A[],
+  id: string
+): A | null {
+  return revisions.find(x => getRevisionId(x) === id) ?? null
+}
+
+export function getRevisionId(revised: RevisedSpec): string {
+  const year = revised.revision.getFullYear()
+  const month = revised.revision.getMonth() + 1
+  const day = revised.revision.getDate()
+
+  const paddedMonth = month.toString().padStart(2, '0')
+  const paddedDay = day.toString().padStart(2, '0')
+
+  return `${year}-${paddedMonth}-${paddedDay}`
+}
+
+export function getRevisions(
+  config: RevisedConfig,
+  lang: LanguageCode,
+  revisedType: RevisedType,
+  getTitle: (revisedType: RevisedType) => string = x => titles[x]
+): RevisedPage[] | null {
+  const result = getSpecAndLanguage(config, lang, revisedType)
+
+  if (result === null) {
+    return null
+  } else {
+    const [revisions, lang] = result
+
+    return revisions.map((revision, index) => {
+      return {
+        ...revision,
+        lang,
+        title: getTitle(revisedType),
+        revisedType: revisedType,
+        isCurrentRevision: index === 0
+      }
+    })
+  }
+}
+
+export function getPage(
+  config: UnrevisedConfig,
+  lang: LanguageCode,
+  unrevisedType: UnrevisedType,
+  getTitle: (unrevisedType: UnrevisedType) => string = x => titles[x]
+): Page | null {
+  const result = getSpecAndLanguage(config, lang, unrevisedType)
+
+  return result === null
+    ? null
+    : { ...result[0], lang: result[1], title: getTitle(unrevisedType) }
+}
+
+function getSpecAndLanguage<A extends string, B>(
+  config: Config<A, B>,
+  lang: LanguageCode,
+  kind: A
+): [B, LanguageCode] | null {
```

Comment:
```ts
{
  data: B,
  lang: LanguageCode
}
```

Reply:
And why do you even need to pass `lang` if that's already a function parameter anyway?

Reply:
The returned `lang` is the language of the spec for the static page (which might be the given `lang` or the default language in case the spec in the given language does not exist)